include/orc/dwarf_structs.hpp

```diff
@@ -167,6 +168,99 @@ inline bool operator!=(const attribute& x, const attribute& y) { return !(x == y
 
 std::ostream& operator<<(std::ostream& s, const attribute& x);
 
+/**************************************************************************************************/
+// I'm not a fan of this name.
+struct attribute_sequence {
+    using attributes_type = std::vector<attribute>;
+    using value_type = typename attributes_type::value_type;
+    using iterator = typename attributes_type::iterator;
+    using const_iterator = typename attributes_type::const_iterator;
+
+    void reserve(std::size_t size) {
+        _attributes.reserve(size);
+    }
+
+    bool has(dw::at name) const {
+        auto [valid, iterator] = find(name);
+        return valid;
+    }
+
+    bool has(dw::at name, enum attribute_value::type t) const {
+        auto [valid, iterator] = find(name);
+        return valid ? iterator->has(t) : false;
+    }
+
+    bool has_uint(dw::at name) const {
+        return has(name, attribute_value::type::uint);
+    }
+
+    bool has_string(dw::at name) const {
+        return has(name, attribute_value::type::string);
+    }
+
+    bool has_reference(dw::at name) const {
+        return has(name, attribute_value::type::reference);
+    }
+
+    auto& get(dw::at name) {
+        auto [valid, iterator] = find(name);
+        assert(valid);
+        return *iterator;
+    }
+
+    const auto& get(dw::at name) const {
+        auto [valid, iterator] = find(name);
+        assert(valid);
+        return *iterator;
+    }
+
+    auto hash(dw::at name) const {
```

Comment:
I know this is a little picky, but I'm not a huge fan of `auto` when one could use the simple type: std::string, uint32_t, etc. (Hash and some of the following functions)

Reply:
I can make the change. The advantage here is that the return type depends on the return type of another routine, and `auto` keeps them in sync automatically.